x-pack/plugin/mapper-counted-keyword/src/main/java/org/elasticsearch/xpack/countedkeyword/CountedTermsAggregator.java

```diff
@@ -0,0 +1,177 @@
+    protected final ValuesSource.Bytes.WithOrdinals valuesSource;
+
+    @SuppressWarnings("this-escape")
+    CountedTermsAggregator(
+        String name,
+        AggregatorFactories factories,
+        ValuesSource.Bytes.WithOrdinals valuesSource,
+        BucketOrder order,
+        DocValueFormat format,
+        BucketCountThresholds bucketCountThresholds,
+        AggregationContext context,
+        Aggregator parent,
+        CardinalityUpperBound cardinality,
+        Map<String, Object> metadata
+    ) throws IOException {
+        super(name, factories, context, parent, bucketCountThresholds, order, format, SubAggCollectionMode.DEPTH_FIRST, metadata);
+        this.valuesSource = valuesSource;
+        this.bucketOrds = BytesKeyedBucketOrds.build(context.bigArrays(), cardinality);
+    }
+
+    @Override
+    public LeafBucketCollector getLeafCollector(AggregationExecutionContext aggCtx, LeafBucketCollector sub) throws IOException {
+        SortedSetDocValues ords = valuesSource.ordinalsValues(aggCtx.getLeafReaderContext());
+        return new LeafBucketCollectorBase(sub, ords) {
+
+            @Override
+            public void collect(int doc, long owningBucketOrd) throws IOException {
+                if (ords.advanceExact(doc) == false) {
+                    return;
+                }
+                for (long ord = ords.nextOrd(); ord != NO_MORE_ORDS; ord = ords.nextOrd()) {
+                    long bucketOrdinal = bucketOrds.add(owningBucketOrd, ords.lookupOrd(ord));
+                    if (bucketOrdinal < 0) { // already seen
+                        bucketOrdinal = -1 - bucketOrdinal;
+                        collectExistingBucket(sub, doc, bucketOrdinal);
+                    } else {
+                        collectBucket(sub, doc, bucketOrdinal);
+                    }
+                }
+            }
+        };
+    }
+
+    @Override
+    public InternalAggregation[] buildAggregations(long[] owningBucketOrds) throws IOException {
+        StringTerms.Bucket[][] topBucketsPerOrd = new StringTerms.Bucket[owningBucketOrds.length][];
+        long[] otherDocCounts = new long[owningBucketOrds.length];
+        for (int ordIdx = 0; ordIdx < owningBucketOrds.length; ordIdx++) {
+            int size = (int) Math.min(bucketOrds.size(), bucketCountThresholds.getShardSize());
+
+            BucketPriorityQueue<StringTerms.Bucket> ordered = new BucketPriorityQueue<>(size, partiallyBuiltBucketComparator);
+            StringTerms.Bucket spare = null;
+            BytesKeyedBucketOrds.BucketOrdsEnum ordsEnum = bucketOrds.ordsEnum(owningBucketOrds[ordIdx]);
+            Supplier<StringTerms.Bucket> emptyBucketBuilder = () -> new StringTerms.Bucket(new BytesRef(), 0, null, false, 0, format);
+            while (ordsEnum.next()) {
+                long docCount = bucketDocCount(ordsEnum.ord());
+                otherDocCounts[ordIdx] += docCount;
+                if (docCount < bucketCountThresholds.getShardMinDocCount()) {
```

Comment:
I think this if statement can also be removed here. The doc count parameter isn't exposed in the builder  and anything with a match (doc count of 1) will be included. No need to do additional check here.